railties/lib/rails/rack/silence_request.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# :markup: markdown
+
+require "active_support/logger_silence"
+
+module Rails
+  module Rack
+    # Allows you to silence requests made to a specific path.
+    # This is useful for preventing recurring requests like healthchecks from clogging the logging.
+    # This middleware is used to do just that against the path /up in production by default.
+    #
+    # Example:
+    #
+    #   config.middleware.insert_before \
+    #     Rails::Rack::Logger, Rails::Rack::SilenceRequest, path: "/up"
+    #
+    # This middleware can also be configured using `config.silence_healthcheck = "/up"` in Rails.
+    class SilenceRequest
+      def initialize(app, path:)
+        @app, @path = app, path
+      end
+
+      def call(env)
+        request = ActionDispatch::Request.new(env)
+
+        if request.path == @path
```

Comment:
```suggestion
        if request.path === @paths
```

What would you think about making it a configurable array of paths (String/Regex)? Then we could do something like:

```
config.silent_request_paths << "/assets"
```

in Sprockets/Propshaft instead of having the QuietAssets middleware be almost a copy of this one.

Reply:
I had that at first, but I then arrested myself with a charge of YAGNI. I couldn't come up with any other use cases except for the healthcheck, so decided to focus on that. The API is a lot more straight forward than something generic.

Reply:
But I still think we could reuse this. We could inherit the QuietAssets middleware in Propshaft from this and just hardset the path:. But not sure it's even worth it. The implementation is so small here.

Reply:
In fact, I just made the implementation even slimmer by borrowing from Propshaft's quiet asset middleware. So thanks for the reminder of that one!

Reply:
I use ActiveStorage with the disk controller in dev and the amount of logs I get from that are astronomical, so it would be appreciated if I was able to configure this middleware for that.

I see that it is very slim so I'll probably just joink it for myself if needed but I do believe there is a need here: https://stackoverflow.com/questions/52147873/mute-activestorage-logs